src/kmedoids_algorithm.cpp

```diff
@@ -171,6 +171,7 @@ std::string km::KMedoids::getAlgorithm() {
  */
 void km::KMedoids::setAlgorithm(const std::string& new_alg) {
   algorithm = new_alg;
+  km::KMedoids::checkAlgorithm(algorithm);
```

Comment:
Good change; we should be checking the algorithm here.

However I don't think we need the temporary variable `algorithm` -- we can just pass `new_alg` directly, because I don't think that memory will ever be modified

Reply:
algorithm is a global variable (yes, it is very confusing...) so we need to update it whenever the user sets a new algorithm. Then we check if the new algorithm is valid or not. An error is thrown if it is not valid.

Reply:
Ok I see; so this PR doesn't fully resolve #110 , but instead makes the `this->` explicit.